From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Wed, 8 Jun 2022 18:28:13 +0100
Subject: ensure grub-mkconfig ignores *.apk-new files

If one of the Grub template files in /etc/grub.d/ is locally modified
then whenever an updated version of the Alpine Grub package is next
installed apk will place a stock version of the modified file in the
same directory with a .apk-new suffix. Whenever grub-mkconfig is next
triggered to regenerate grub.cfg it will use both the locally modified
file and the stock file which will result in duplicate Grub menu
entries.

Modify grub-mkconfig to ignore *.apk-new files.

---

diff -aur a/util/grub-mkconfig.in b/util/grub-mkconfig.in
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -278,6 +278,10 @@
     *~) ;;
     # emacsen autosave files. FIXME: support other editors
     */\#*\#) ;;
+    # Copy of a template file from Alpine Grub package where the stock file
+    # was locally modified. Any such *.apk-new files should be ignored by
+    # grub-mkconfig.
+    *.apk-new) ;;
     *)
       if grub_file_is_not_garbage "$i" && test -x "$i" ; then
         echo
